builtin/credential/aws/path_role.go

```diff
@@ -320,6 +322,142 @@ func (b *backend) setRole(ctx context.Context, s logical.Storage, roleName strin
 	return nil
 }
 
+// initialize is used to initialize the AWS roles
+func (b *backend) initialize(ctx context.Context, s logical.Storage) error {
+
+	// Initialize only if we are either:
+	//   (1) A local mount.
+	//   (2) Are _not_ a replicated standby cluster.
+	if b.System().LocalMount() || !b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary|consts.ReplicationPerformanceStandby) {
+
+		logger := b.Logger().Named("initialize")
+
+		// grab the guard
+		if !atomic.CompareAndSwapUint32(b.initializeCASGuard, 0, 1) {
+			logger.Error("initialization already in progress")
+			return nil
+		}
+
+		// check if this method has already been called
+		if b.isInitialized {
+			atomic.StoreUint32(b.initializeCASGuard, 0)
+			logger.Error("already initialized")
+			return nil
+		}
+		b.isInitialized = true
+
+		// kick off the role upgrader
+		go func() {
```

Comment:
Why is this running in a goroutine? This turns the synchronous function into asynchronous which means we'll return to core immediately, That means it will start serving client requests immediately even though we want to ensure this runs before client requests start as that's the guarantee of the initialization function.